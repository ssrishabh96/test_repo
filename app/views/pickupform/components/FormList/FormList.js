import React, { Component } from 'react';
// import { groupBy, compose, map, evolve, zipObj, toPairs, pathOr, tap } from 'ramda';

import { View, Text } from 'react-native';
import { KeyboardAwareSectionList } from 'react-native-keyboard-aware-scroll-view';
import PropTypes from 'prop-types';

import LabelItem from '../FormItems/labelItem';
import TextItem from '../FormItems/textItem';
import VinItem from '../FormItems/vinItem';
import ButtonsItem from '../FormItems/buttonsItem';
import MultiImageItem from '../FormItems/multiImageItem';
import ImageItem from '../FormItems/imageItem';
import CommentsItem from '../FormItems/commentsItem';
import DateItem from '../FormItems/dateItem';
import AccessoryItem from '../FormItems/accessoryItem';
import SignatureItem from '../FormItems/signatureItem';
import colors from '../../../../styles/colors';

import { DataSource, ReferenceCodes } from '../../pickupformTypes';

export default class FormList extends Component {
  static propTypes = {
    handleOnSelect: PropTypes.func.isRequired,
    data: PropTypes.array.isRequired, // eslint-disable-line
    onRef: PropTypes.func.isRequired,
    openPicker: PropTypes.func.isRequired,
    openCamera: PropTypes.func.isRequired,
    handlePreviewClick: PropTypes.func.isRequired,
    showWarningNotification: PropTypes.func.isRequired,
    showSignatureView: PropTypes.func.isRequired,
    showScannerView: PropTypes.func.isRequired,
    updatePickupLotData: PropTypes.func.isRequired,
    referenceCodes: PropTypes.shape(ReferenceCodes).isRequired,
  };
  componentDidMount() {
    this.props.onRef(this.sectionref);
  }
  componentWillUnmount() {
    this.props.onRef(undefined);
  }
  renderItem = (item) => {
    if (!item.item.readable) return false;
    switch (item.item.type) {
      case 'label':
        return (
          <LabelItem
            key={item.item.field}
            {...item.item}
            currentPickupLot={this.props.currentPickupLot}
          />
        );
      case 'accessory':
        return (
          <AccessoryItem
            key={item.item.field}
            {...item.item}
            value={this.props.currentPickupLot[item.item.key1]}
            currentPickupLot={this.props.currentPickupLot}
            handleOnSelect={this.handleOnSelect}
            referenceCodes={this.props.referenceCodes}
            openPicker={this.props.openPicker}
            showWarningNotification={this.props.showWarningNotification}
          />
        );
      case 'text':
        return (
          <TextItem
            key={item.item.field}
            {...item.item}
            value={this.props.currentPickupLot[item.item.key1]}
            handleOnSelect={this.props.handleOnSelect}
            showWarningNotification={this.props.showWarningNotification}
          />
        );
      case 'buttons':
        return (
          <ButtonsItem
            key={item.item.field}
            {...item.item}
            value={this.props.currentPickupLot[item.item.key1]}
            currentPickupLot={this.props.currentPickupLot}
            handleOnSelect={this.props.handleOnSelect}
            showWarningNotification={this.props.showWarningNotification}
          />
        );
      case 'comments':
        return (
          <CommentsItem
            key={item.item.field}
            {...item.item}
            value={this.props.currentPickupLot[item.item.key1]}
            handleOnSelect={this.props.handleOnSelect}
            showWarningNotification={this.props.showWarningNotification}
          />
        );
      case 'date':
        return (
          <DateItem
            key={item.item.field}
            {...item.item}
            value={this.props.currentPickupLot[item.item.key1]}
            handleDateChange={this.props.handleOnSelect}
            showWarningNotification={this.props.showWarningNotification}
          />
        );
      case 'signature':
        return (
          <SignatureItem
            key={item.item.field}
            {...item.item}
            value={this.props.currentPickupLot[item.item.key1]}
            handleOnSelect={this.props.handleOnSelect}
            showSignatureView={this.props.showSignatureView}
            showWarningNotification={this.props.showWarningNotification}
          />
        );
      case 'multiImage':
        return (
          <MultiImageItem
            key={item.item.field}
            {...item.item}
            value={this.props.currentPickupLot[item.item.key1]}
            handleOnSelect={this.props.handleOnSelect}
            openCamera={this.props.openCamera}
            handlePreviewClick={this.props.handlePreviewClick}
            showWarningNotification={this.props.showWarningNotification}
            documentType={item.item.documentType}
          />
        );
      case 'image':
        return (
          <ImageItem
            key={item.item.field}
            {...item.item}
            value={this.props.currentPickupLot[item.item.key1]}
            handleOnSelect={this.props.handleOnSelect}
            openCamera={this.props.openCamera}
            handlePreviewClick={this.props.handlePreviewClick}
            showWarningNotification={this.props.showWarningNotification}
            documentType={item.item.documentType}
          />
        );
      case 'vin':
        return (
          <VinItem
            key={item.item.field}
            {...item.item}
            value={this.props.currentPickupLot[item.item.key1]}
            handleOnSelect={this.props.handleOnSelect}
            showScannerView={this.props.showScannerView}
            showWarningNotification={this.props.showWarningNotification}
          />
        );
      case 'bottomBuffer':
        return <View style={{ height: 90 }} />;
      default:
        return null;
    }
  };

  renderHeader = headerItem => (
    <Text
      style={[
        {
          padding: 8,
          fontSize: 14,
          fontWeight: 'bold',
          color: 'white',
          backgroundColor: colors.SECTION_HEADER_BG,
        },
      ]}
    >
      {headerItem.section.key.toUpperCase()}
    </Text>
  );

  render() {
    const { data } = this.props;
    if (!data) {
      return (
        <View>
          <Text>Loading...</Text>
        </View>
      );
    }
    return (
      <KeyboardAwareSectionList
        innerRef={(ref) => {
          this.sectionref = ref;
        }}
        renderItem={this.renderItem}
        renderSectionHeader={this.renderHeader}
        sections={data}
      />
    );
  }
}
