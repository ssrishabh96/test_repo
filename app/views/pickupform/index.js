import PickupForm from './pickupform.view';

export default PickupForm;
