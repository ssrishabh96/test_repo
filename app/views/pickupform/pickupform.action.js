import { propOr, compose, prop, defaultTo, reject, pathOr, propEq } from 'ramda';
import moment from 'moment';
// import dbquery from './database/dbquery';
import {
  PICK_FORM_SET_CURRENT_LOT,
  PICK_FORM_CLEAR_CURRENT_LOT,
  PICK_FORM_CACHE_LOT,
  PICK_FORM_CACHE_IMAGE,
  PICK_FORM_REMOVE_FROM_CACHE,
  PICK_FORM_STEP_INCREMENT,
  PICK_FORM_STEP_DECREMENT,
  PICK_FORM_RESET_STEP,
  PICK_FORM_UPDATE_LOT_DATA,
  PICK_FORM_UPDATE_WARNING,
  PICK_FORM_IMAGE_PREVIEW,
  PICK_FORM_SUBMITTING,
  PICK_FORM_SUBMIT_STATUS,
  PICK_FORM_SUBMIT_COMPLETE,
  FORM_GOTO_STEP,
} from './pickupform.constants';
import { LOTLIST_LOADED } from 'views/lotlist/lotlist.constants';
import { uploadImage, postPickupForm } from './pickupform.service';

import { removeLotInfoFromCache } from '../lotview/lotview.action';

import { getCacheForCurrentUser } from './pickupform.redux';
import { getActiveVendorId } from 'views/login/helpers/loginHelpers';
import { buildImage, buildMultiImage, prepareFormData } from './pickupform.helper';
import { getLotCharges } from 'views/lotview/lotview.service';

import { buildErrorMessage } from 'utils/mappers/errorMessageMapper';
import { getLotTripFormByType } from 'constants/tripTypeMap';

// prefils the vehicleDescCorrection with vehicleDescription
// while still mapped to different keys so changeing one does not change the other
const defaultToEmpty = defaultTo('');
const getAddress = (location) => {
  const address = `${defaultToEmpty(location.line_1)} ${defaultToEmpty(
    location.city,
  )} ${defaultToEmpty(location.state)} ${defaultToEmpty(location.zip)}`;
  return /^\s+$/.test(address) ? '' : address;
};

const byTripType = (lot) => {
  if (lot.tripType === 'P') {
    return {
      pickUpAddress: getAddress(lot.source),
      yardAddress: getAddress(lot.destination),
    };
  }
  return {
    deliveryAddress: getAddress(lot.destination),
    yardAddress: getAddress(lot.source),
  };
};
const preparePickupLot = lot => ({
  ...lot,
  ...byTripType(lot),
  vehicleDescCorrection: propOr('', 'description')(lot),
  // driverPrint: propOr('', 'responsible_party_name')(lot),
});

export const setCurrentPickupLot = (lot, charges = {}, restart = false) => (dispatch, getState) => {
  const cache = restart
    ? {}
    : propOr({}, lot.dispatch_assignment_detail_id)(getCacheForCurrentUser(getState()));
  const preparedLot = cache.lot
    ? { ...cache.lot, ...charges }
    : preparePickupLot({
      otherRefusedSign: 'No',
      transportDate: moment().format('MM/DD/YY'),
      datePickedUp: moment().format('MM/DD/YY'),
      creditCardDate: moment().format('MM/DD/YY'),
      checkTotal: `${charges.total}`,
      cashTotal: `${charges.total}`,
      creditCardTotal: `${charges.total}`,
      ...lot,
      ...charges,
    });
  console.log('?? setcurrent pickup lot', { lot, charges, preparedLot });
  const steps = JSON.parse(JSON.stringify(getLotTripFormByType(preparedLot).formSteps));
  for (let s = 0; s < steps.length; s += 1) {
    for (let i = 0; i < steps[s].vehicleDetails.length; i += 1) {
      steps[s].vehicleDetails[i].value = preparedLot[steps[s].vehicleDetails[i].key1] || '';
    }
  }
  return dispatch({
    type: PICK_FORM_SET_CURRENT_LOT,
    currentPickupLot: preparedLot,
    steps,
  });
};
export const clearCurrentPickupLot = () => ({
  type: PICK_FORM_CLEAR_CURRENT_LOT,
});

export const cacheCurrentForm = (lot, isAwaitingSync = false) => (dispatch, getState) =>
  dispatch({
    type: PICK_FORM_CACHE_LOT,
    id: lot.dispatch_assignment_detail_id,
    userId: getActiveVendorId(getState()),
    lot,
    isAwaitingSync,
  });
export const removeFormFromCache = ({ dispatch_assignment_detail_id }) => (dispatch, getState) =>
  // const userId = getActiveVendorId(getState());
  // const images = getState().pickUpForm.cache[userId][dispatch_assignment_detail_id].images;
  // console.log(images);
  dispatch({
    type: PICK_FORM_REMOVE_FROM_CACHE,
    id: dispatch_assignment_detail_id,
    userId: getActiveVendorId(getState()),
  });
export const setImageInCache = (lot, fileName, reqId) => (dispatch, getState) =>
  dispatch({
    type: PICK_FORM_CACHE_IMAGE,
    id: lot.dispatch_assignment_detail_id,
    userId: getActiveVendorId(getState()),
    fileName,
    reqId,
  });

export const incrementCurrentStep = () => ({ type: PICK_FORM_STEP_INCREMENT });

export const decrementCurrentStep = () => ({ type: PICK_FORM_STEP_DECREMENT });

export const resetCurrentStep = () => ({ type: PICK_FORM_RESET_STEP });

export const goToStep = step => ({ type: FORM_GOTO_STEP, step });

export const updatePickupLotData = (key, value) => ({
  type: PICK_FORM_UPDATE_LOT_DATA,
  key,
  value,
});

export const updateWarningFlags = (items, value) => ({
  type: PICK_FORM_UPDATE_WARNING,
  items,
  value,
});
export const setCurrentItemValue = () => (dispatch) => {
  //   dispatch({ type: RESET_MULTISELECT_MODE });
  //   dispatch({ type: CLEAR_SELECTION });
};
export const showHideImagePreview = (value: boolean, imagePreview: Object) => ({
  type: PICK_FORM_IMAGE_PREVIEW,
  value,
  imagePreview,
});

export const syncFormSubmission = (dispatch_assignment_detail_id, navigator) => (
  dispatch,
  getState,
) => {
  // console.log('in sync...');
  const lot = pathOr({}, [dispatch_assignment_detail_id, 'lot'])(
    getCacheForCurrentUser(getState()),
  );
  if (!lot) return Promise.reject({ message: 'something went wrong... lot was not in cache...' });
  // console.log('got the lot');
  return getLotCharges(dispatch_assignment_detail_id)
    .then(({ charges }) => {
      // console.log('got the charges', { charges });
      const newLot = { ...lot, ...charges };
      // console.log({ newLot });
      const data = { navigator, props: { navigator } }; // TODO fix this madness
      return dispatch(uploadPickupForm(newLot, data));
    })
    .catch((e) => {
      console.log('something went wrong...', e);
      Promise.reject({ message: 'getLotCharges probably failed' });
    });
};
export const setFormSubmitting = status => ({
  type: PICK_FORM_SUBMITTING,
  status,
});
export const setFormSubmittingStatus = status => ({
  type: PICK_FORM_SUBMIT_STATUS,
  status,
});
export const setFormSubmittingComplete = () => ({
  type: PICK_FORM_SUBMIT_COMPLETE,
});
export const uploadPickupForm = (currentPickupLot, data) => (dispatch, getState) => {
  console.log('uploading the form');
  // TODO if no network, dispatch(CACHE_LOT_WAITING_FOR_SYNC); return;
  dispatch(setFormSubmitting('Preparing form for upload.'));
  const uploadAttachment =
    (currentPickupLot.uploadAttachment && buildMultiImage(currentPickupLot.uploadAttachment)) || [];
  const attachment =
    (currentPickupLot.attachment && buildMultiImage(currentPickupLot.attachment)) || [];
  let filePaths = uploadAttachment.concat(attachment);
  if (currentPickupLot.creditCardImage) {
    filePaths = filePaths.concat(buildMultiImage(currentPickupLot.creditCardImage));
  }
  if (currentPickupLot.checkImage) {
    filePaths = filePaths.concat(buildMultiImage(currentPickupLot.checkImage));
  }
  if (currentPickupLot.vehicleImages) {
    filePaths.push(buildImage(currentPickupLot.vehicleImages));
  }
  const imageInfo = {
    lotNumber: currentPickupLot.number,
    yard: currentPickupLot.yardNumber,
    filePaths,
  };
  const files = imageInfo.filePaths || [];
  return uploadImages(files, currentPickupLot, data, dispatch, getState)
    .then((lot) => {
      dispatch(setFormSubmittingStatus('Uploading Form'));
      const sendData = prepareFormData(lot);
      return postPickupForm({
        assignmentId: lot.dispatch_assignment_detail_id,
        data: sendData,
      });
    })
    .then((response) => {
      if (response.status === 200 && response.data.status === 'success') {
        dispatch(setFormSubmittingStatus('Upload Complete.'));

        const lotList = pathOr([], ['lotlist', 'inProgress', 'lots'])(getState());
        const updatedLotList = reject(propEq('number', currentPickupLot.number))(lotList);
        dispatch({ type: LOTLIST_LOADED, bucket: 'inProgress', lots: updatedLotList });
        const messages = {
          D: 'Form submitted Successfully.\nLot is moved to completed',
          P: 'Form submitted Successfully.\nLot is moved to in transit',
        };
        setTimeout(
          () =>
            data.props.navigator.showInAppNotification({
              screen: 'CopartTransporter.ShowInAppNotification',
              passProps: {
                type: 'success',
                content: messages[currentPickupLot.formType.toUpperCase()],
              },
              autoDismissTimerSec: 3,
            }),
          400,
        );
        dispatch(resetCurrentStep());
        setTimeout(() => {
          dispatch(setFormSubmittingComplete());
        }, 2000);
        dispatch(removeLotInfoFromCache(currentPickupLot.dispatch_assignment_detail_id));
        dispatch(removeFormFromCache(currentPickupLot));
      } else {
        console.log('Submit form data failed', response);
        const errorMessage = response.errors ? response.errors : response.data.error;
        return Promise.reject(response.data);
      }
    })
    .catch((err) => {
      console.log('caught form upload error', err);
      const errorsData = pathOr(err, ['response', 'data'])(err);
      const errorMessage = buildErrorMessage(errorsData);
      dispatch(setFormSubmittingComplete());
      dispatch(cacheCurrentForm(currentPickupLot, true));
      setTimeout(
        () =>
          data.props.navigator.showInAppNotification({
            screen: 'CopartTransporter.ShowInAppNotification',
            passProps: {
              type: 'error',
              content: `Form Saved. Uploading  Failed. please try again\n ${errorMessage || ''}`,
              autoDismissTimerSec: 15,
            },
          }),
        400,
      );
      return Promise.reject(err);
    });
};

const uploadImages = (files, currentPickupLot, data, dispatch, getState, index = 0) => {
  if (index >= files.length || !files[index]) {
    dispatch(setFormSubmittingStatus('Image upload complete.'));
    return Promise.resolve(currentPickupLot);
  }
  dispatch(setFormSubmittingStatus(`Uploading image... ${index + 1} of ${files.length}`));
  const element = files[index];
  const reqID = pathOr(false, [
    currentPickupLot.dispatch_assignment_detail_id,
    'images',
    element.filename,
  ])(getCacheForCurrentUser(getState()));
  if (reqID) {
    const newLot = setImageOnLotWithRequestID(currentPickupLot, reqID, element);
    return uploadImages(files, newLot, data, dispatch, getState, index + 1);
  }
  return uploadImage(element, currentPickupLot.dispatch_assignment_detail_id, index)
    .then((result) => {
      if (result && result.status === 200) {
        const parseResponse = parseJSON(result.data);
        const newLot = setImageOnLotWithRequestID(
          currentPickupLot,
          parseResponse.request_id,
          element,
        );
        dispatch(setImageInCache(newLot, files[index].filename, parseResponse.request_id));
        return uploadImages(files, newLot, data, dispatch, getState, index + 1);
      }
      // console.log('Upload Failed', { result });
      return Promise.reject({ result });
    })
    .catch((err) => {
      console.log('Image upload encountered an error...', err);
      return Promise.reject(err);
    });
};

function parseJSON(param) {
  try {
    return JSON.parse(param);
  } catch (ex) {
    return param;
  }
}

const setImageOnLotWithRequestID = (lot, reqID, element) => {
  const newLot = { ...lot };
  const newObject = element.value;
  if (element.type === 'image') {
    newObject.requestId = reqID;
    newLot[newObject.key1] = newObject;
  } else {
    const a = newObject.filter(x => x.image.pathName.replace('file://', '') === element.filepath);
    a[0].image.requestId = reqID;
    newLot[newObject.key1] = newObject;
  }
  return newLot;
};
