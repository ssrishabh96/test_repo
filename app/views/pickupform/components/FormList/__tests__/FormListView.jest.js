import React from 'react';
import { shallow } from 'enzyme';

import FormList from '../';
import StepOne from '../../../pickupsteps/step1.view';
import { convertData } from '../../../pickupform.view';
import referenceCodes from 'utils/mappers/referenceCodes';

const defaultLot = {
  dispatch_assignment_detail_id: 42,
  tripType: 'P',
  formType: 'P',
  source: { line_1: 123, city: 'plano', state: 'TX', zip: 75075 },
  destination: { line_1: null, city: null, state: null, zip: null },
  description: 'Blue Honda',
  vehicleDescCorrection: 'Blue Honda',
  yardAddress: '',
  pickUpAddress: '123 plano TX 75075',
};
const defaultProps = {
  data: convertData(StepOne),
  currentPickupLot: defaultLot,
  handleOnSelect: jest.fn(),
  onRef: jest.fn(),
  openPicker: jest.fn(),
  openCamera: jest.fn(),
  handlePreviewClick: jest.fn(),
  showWarningNotification: jest.fn(),
  showSignatureView: jest.fn(),
  showScannerView: jest.fn(),
  updatePickupLotData: jest.fn(),
  referenceCodes,
};

test('it renders', () => {
  const wrapper = shallow(<FormList {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});
