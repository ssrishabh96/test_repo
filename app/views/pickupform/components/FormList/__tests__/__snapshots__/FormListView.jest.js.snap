// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<_class
  enableAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  innerRef={[Function]}
  keyboardOpeningTime={250}
  renderItem={[Function]}
  renderSectionHeader={[Function]}
  sections={
    Array [
      Object {
        "data": Array [
          Object {
            "editable": false,
            "field": "vehicleDescription",
            "id": "1",
            "key1": "description",
            "label": "Vehicle Description",
            "readable": true,
            "required": false,
            "section": "Vehicle Information",
            "showWarning": false,
            "type": "label",
            "value": "",
          },
          Object {
            "editable": false,
            "field": "damageType",
            "id": "2",
            "key1": "lossType",
            "label": "Damage Type",
            "readable": true,
            "required": false,
            "section": "Vehicle Information",
            "showWarning": false,
            "type": "label",
            "value": "",
          },
          Object {
            "editable": false,
            "field": "licenseNumber",
            "id": "3",
            "key1": "licenseNumber",
            "label": "License Number",
            "readable": true,
            "required": false,
            "section": "Vehicle Information",
            "showWarning": false,
            "type": "label",
            "value": "",
          },
          Object {
            "editable": false,
            "field": "vin",
            "id": "4",
            "key1": "vin",
            "label": "VIN",
            "readable": true,
            "required": false,
            "section": "Vehicle Information",
            "showWarning": false,
            "type": "label",
            "value": "",
          },
          Object {
            "editable": false,
            "field": "vehicleType",
            "id": "5",
            "key1": "vehicleType",
            "label": "Vehicle Type",
            "readable": true,
            "required": false,
            "section": "Vehicle Information",
            "showWarning": false,
            "type": "label",
            "value": "",
          },
          Object {
            "editable": false,
            "field": "driverName",
            "id": "6",
            "key1": "responsible_party_name",
            "label": "Driver Name (Tow Provider)",
            "readable": true,
            "required": false,
            "section": "Vehicle Information",
            "showWarning": false,
            "type": "label",
            "value": "",
          },
          Object {
            "editable": false,
            "field": "isTowable",
            "id": "7",
            "key1": "isTowable",
            "label": "Is Tow-able",
            "readable": true,
            "required": false,
            "section": "Vehicle Information",
            "showWarning": false,
            "type": "label",
            "value": "",
          },
        ],
        "key": "Vehicle Information",
      },
      Object {
        "data": Array [
          Object {
            "editable": false,
            "field": "pickUpAddress",
            "id": "8",
            "key1": "pickUpAddress",
            "label": "Pickup Address",
            "readable": true,
            "required": false,
            "section": "Location Information",
            "showWarning": false,
            "type": "label",
            "value": "",
          },
          Object {
            "editable": false,
            "field": "yardAddress",
            "id": "9",
            "key1": "yardAddress",
            "label": "Yard Address",
            "readable": true,
            "required": false,
            "section": "Location Information",
            "showWarning": false,
            "type": "label",
            "value": "",
          },
          Object {
            "editable": false,
            "field": "tripZone",
            "id": "10",
            "key1": "tripZone",
            "label": "Trip Zone",
            "readable": true,
            "referenceDataKey": "other",
            "required": false,
            "section": "Location Information",
            "showWarning": false,
            "type": "label",
            "value": "",
          },
          Object {
            "editable": false,
            "field": "advanceLimit",
            "id": "11",
            "key1": "advanceLimit",
            "label": "Advance Limit",
            "readable": true,
            "required": false,
            "section": "Location Information",
            "showWarning": false,
            "type": "label",
            "value": "",
          },
          Object {
            "editable": false,
            "field": "vehicleWeight",
            "id": "12",
            "key1": "vehicle_weight",
            "label": "Vehicle Weight",
            "readable": true,
            "required": false,
            "section": "Location Information",
            "showWarning": false,
            "type": "label",
            "value": "",
          },
        ],
        "key": "Location Information",
      },
      Object {
        "data": Array [
          Object {
            "editable": true,
            "field": "puTitle",
            "id": "13",
            "key1": "puTitle",
            "label": "Pickup Title",
            "optionKeys": "Yes|No",
            "readable": true,
            "required": true,
            "section": "Documents Collected",
            "showWarning": false,
            "type": "buttons",
            "value": "",
          },
          Object {
            "editable": true,
            "field": "lienRelease",
            "id": "14",
            "key1": "lienRelease",
            "label": "Get Lien Release",
            "optionKeys": "Yes|No",
            "readable": true,
            "required": true,
            "section": "Documents Collected",
            "showWarning": false,
            "type": "buttons",
            "value": "",
          },
          Object {
            "editable": true,
            "field": "poa",
            "id": "15",
            "key1": "poa",
            "label": "Get P.O.A",
            "optionKeys": "Yes|No",
            "readable": true,
            "required": true,
            "section": "Documents Collected",
            "showWarning": false,
            "type": "buttons",
            "value": "",
          },
          Object {
            "autoCapitalize": "sentences",
            "editable": true,
            "field": "otherDocuments",
            "id": "16",
            "key1": "otherDocuments",
            "label": "Other",
            "readable": true,
            "required": false,
            "section": "Documents Collected",
            "showWarning": false,
            "type": "text",
            "validations": Object {
              "maxLength": 20,
            },
            "value": "",
          },
        ],
        "key": "Documents Collected",
      },
      Object {
        "data": Array [
          Object {
            "autoCapitalize": "sentences",
            "editable": true,
            "field": "vehicleDescCorrection",
            "id": "17",
            "key1": "vehicleDescCorrection",
            "label": "Vehicle Description Correction",
            "readable": true,
            "required": false,
            "section": "Vehicle Details",
            "showWarning": false,
            "type": "text",
            "validations": Object {
              "maxLength": 30,
            },
            "value": "",
          },
          Object {
            "editable": true,
            "field": "towType",
            "id": "18",
            "key1": "towType",
            "label": "Tow Type ?",
            "optionKeys": "Standard|Medium|Heavy",
            "optionValues": "S|M|H",
            "readable": true,
            "required": false,
            "section": "Vehicle Details",
            "showWarning": false,
            "type": "buttons",
            "value": "",
          },
          Object {
            "editable": true,
            "field": "keys ?",
            "id": "19",
            "key1": "keys",
            "label": "Keys ?",
            "optionKeys": "Yes|No",
            "readable": true,
            "required": true,
            "section": "Vehicle Details",
            "showWarning": false,
            "type": "buttons",
            "value": "",
          },
          Object {
            "editable": true,
            "field": "keyFob",
            "id": "20",
            "key1": "keyFob",
            "label": "Keys Fob ?",
            "optionKeys": "Yes|No",
            "readable": true,
            "required": true,
            "section": "Vehicle Details",
            "showWarning": false,
            "type": "buttons",
            "value": "",
          },
          Object {
            "editable": true,
            "field": "vehicleColor",
            "id": "21",
            "key1": "colors",
            "label": "Vehicle Color",
            "multiselect": false,
            "readable": true,
            "referenceDataKey": "colors",
            "required": true,
            "section": "Vehicle Details",
            "showWarning": false,
            "type": "accessory",
            "value": "",
          },
          Object {
            "editable": true,
            "field": "primaryDamage",
            "id": "22",
            "key1": "primaryDamage",
            "label": "Primary Damage",
            "multiselect": false,
            "readable": true,
            "referenceDataKey": "damageCodes",
            "required": true,
            "section": "Vehicle Details",
            "showWarning": false,
            "type": "accessory",
            "value": "",
          },
          Object {
            "editable": true,
            "field": "secondaryDamage",
            "id": "23",
            "key1": "secondaryDamage",
            "label": "Secondary Damage",
            "multiselect": false,
            "readable": true,
            "referenceDataKey": "damageCodes",
            "required": true,
            "section": "Vehicle Details",
            "showWarning": false,
            "type": "accessory",
            "value": "",
          },
          Object {
            "autoCapitalize": "characters",
            "editable": true,
            "field": "licensePlate",
            "id": "24",
            "key1": "licensePlate",
            "label": "License Plate",
            "readable": true,
            "required": true,
            "section": "Vehicle Details",
            "showWarning": false,
            "type": "text",
            "validations": Object {
              "maxLength": 10,
            },
            "value": "",
          },
          Object {
            "editable": true,
            "field": "state",
            "id": "25",
            "key1": "state",
            "label": "State",
            "multiselect": false,
            "readable": true,
            "referenceDataKey": "stateCodes",
            "required": true,
            "section": "Vehicle Details",
            "showWarning": false,
            "type": "accessory",
            "value": "",
          },
          Object {
            "editable": true,
            "field": "numberOfPlates",
            "id": "26",
            "key1": "numberOfPlates",
            "label": "Number of Plates",
            "multiselect": false,
            "readable": true,
            "referenceDataKey": "numberOfPlates",
            "required": true,
            "section": "Vehicle Details",
            "showWarning": false,
            "type": "accessory",
            "value": "",
          },
          Object {
            "editable": true,
            "field": "vinPlate",
            "id": "27",
            "key1": "vinPlate",
            "label": "VIN Plate ?",
            "optionKeys": "Yes|No",
            "readable": true,
            "required": true,
            "section": "Vehicle Details",
            "showWarning": false,
            "type": "buttons",
            "value": "",
          },
          Object {
            "editable": true,
            "field": "shopVIN",
            "id": "28",
            "key1": "shopVIN",
            "label": "Shop VIN",
            "readable": true,
            "required": false,
            "section": "Vehicle Details",
            "showWarning": false,
            "type": "vin",
            "validations": Object {
              "maxLength": 17,
            },
            "value": "",
          },
        ],
        "key": "Vehicle Details",
      },
      Object {
        "data": Array [
          Object {
            "editable": true,
            "field": "mechanical",
            "id": "29",
            "key1": "mechanical",
            "label": "Mechanical",
            "multiselect": true,
            "readable": true,
            "referenceDataKey": "mechanicalPartCodes",
            "required": false,
            "section": "Vehicle Missing Parts",
            "showWarning": false,
            "type": "accessory",
            "value": "",
          },
          Object {
            "editable": true,
            "field": "interior",
            "id": "30",
            "key1": "interior",
            "label": "Interior",
            "multiselect": true,
            "readable": true,
            "referenceDataKey": "interiorPartCodes",
            "required": false,
            "section": "Vehicle Missing Parts",
            "showWarning": false,
            "type": "accessory",
            "value": "",
          },
          Object {
            "editable": true,
            "field": "sheetMetal",
            "id": "31",
            "key1": "sheetMetal",
            "label": "Sheet Metal",
            "multiselect": true,
            "readable": true,
            "referenceDataKey": "sheetMetalPartCodes",
            "required": false,
            "section": "Vehicle Missing Parts",
            "showWarning": false,
            "type": "accessory",
            "value": "",
          },
          Object {
            "editable": true,
            "field": "exterior",
            "id": "32",
            "key1": "exterior",
            "label": "Exterior",
            "multiselect": true,
            "readable": true,
            "referenceDataKey": "exteriorPartCodes",
            "required": false,
            "section": "Vehicle Missing Parts",
            "showWarning": false,
            "type": "accessory",
            "value": "",
          },
        ],
        "key": "Vehicle Missing Parts",
      },
      Object {
        "data": Array [
          Object {
            "editable": true,
            "field": "driverSideAir",
            "id": "33",
            "key1": "driverSideAir",
            "label": "Driver Side Airbag",
            "optionKeys": "Damaged|Missing",
            "readable": true,
            "required": false,
            "section": "Airbag Details",
            "showWarning": false,
            "type": "buttons",
            "value": "",
          },
          Object {
            "editable": true,
            "field": "passSideAir",
            "id": "34",
            "key1": "passSideAir",
            "label": "Passenger Side Airbag",
            "optionKeys": "Damaged|Missing",
            "readable": true,
            "required": false,
            "section": "Airbag Details",
            "showWarning": false,
            "type": "buttons",
            "value": "",
          },
          Object {
            "editable": true,
            "field": "leftSideAir",
            "id": "35",
            "key1": "leftSideAir",
            "label": "Left Side Airbag",
            "optionKeys": "Damaged|Missing",
            "readable": true,
            "required": false,
            "section": "Airbag Details",
            "showWarning": false,
            "type": "buttons",
            "value": "",
          },
          Object {
            "editable": true,
            "field": "rightSideAir",
            "id": "36",
            "key1": "rightSideAir",
            "label": "Right Side Airbag",
            "optionKeys": "Damaged|Missing",
            "readable": true,
            "required": false,
            "section": "Airbag Details",
            "showWarning": false,
            "type": "buttons",
            "value": "",
          },
        ],
        "key": "Airbag Details",
      },
      Object {
        "data": Array [
          Object {
            "editable": true,
            "field": "numberOfWheels",
            "id": "37",
            "key1": "numberOfWheels",
            "label": "Number of Wheels",
            "multiselect": false,
            "readable": true,
            "referenceDataKey": "numberOfWheels",
            "required": true,
            "section": "Wheel Details",
            "showWarning": false,
            "type": "accessory",
            "value": "",
          },
          Object {
            "editable": true,
            "field": "wheelType",
            "id": "38",
            "key1": "wheelType",
            "label": "Wheel Type",
            "optionKeys": "Alloy|Steel",
            "readable": true,
            "required": true,
            "section": "Wheel Details",
            "showWarning": false,
            "type": "buttons",
            "value": "",
          },
          Object {
            "editable": true,
            "field": "wheelModel",
            "id": "39",
            "key1": "wheelModel",
            "label": "Wheel Model",
            "optionKeys": "Standard|Custom",
            "readable": true,
            "required": true,
            "section": "Wheel Details",
            "showWarning": false,
            "type": "buttons",
            "value": "",
          },
          Object {
            "editable": true,
            "field": "evidenceOfTeardown",
            "id": "40",
            "key1": "evidenceOfTeardown",
            "label": "Evidence of Teardown",
            "optionKeys": "Yes|No",
            "readable": true,
            "required": true,
            "section": "Wheel Details",
            "showWarning": false,
            "type": "buttons",
            "value": "",
          },
        ],
        "key": "Wheel Details",
      },
      Object {
        "data": Array [
          Object {
            "documentType": "DISPATCH_IMAGE",
            "editable": true,
            "field": "vehicleImages",
            "id": "41",
            "key1": "vehicleImages",
            "label": "Vehicle Image",
            "readable": true,
            "required": true,
            "section": "Vehicle Image",
            "showWarning": false,
            "type": "image",
            "value": "",
          },
        ],
        "key": "Vehicle Image",
      },
    ]
  }
  viewIsInsideTabBar={false}
/>
`;
